zulipterminal/config/themes.py

```diff
@@ -531,3 +537,38 @@ def theme_with_monochrome_added(theme: ThemeSpec) -> ThemeSpec:
             new_style = style
         updated_theme.append(new_style)
     return updated_theme
+
+
+NEW_THEMES = {
+    "gruvbox_dark": gruvbox,
+    "zt_dark": zt_dark,
+    "zt_light": zt_light,
+    "zt_blue": zt_blue,
+}
+
+
+def generate_theme(theme_name: str, color_depth: int) -> ThemeSpec:
+    theme_styles = NEW_THEMES[theme_name].STYLES
```

Comment:
Let's leave this function to focusing on generating a theme from styles, not the theme name - this makes it independent of the way themes are stored (more decoupled) and easier to test - you don't need the complicated patching in the test.

Reply:
I've moved the rest of the function into `parse_themefile()` which is called in `generate_theme()`.
We could also do other color/theme related stuff like the `add_codehilite_pallette` function into `generate_theme` later.

Reply:
This seems fine for now, though I'm not entirely convinced that we shouldn't just have this inline later.